dprof.js

```diff
@@ -100,46 +142,66 @@ Node.prototype.rootIntialize = function () {
   this._before.push(0);
 };
 
-//
-// Setup hooks
-//
-asyncHook.addHooks({
-  init: asyncInit,
-  pre: asyncBefore,
-  post: asyncAfter,
-  destroy: asyncDestroy
-});
-
-const root = new Node({ constructor: { name: 'root' } }, getCallSites(2));
+const root = new Node({}, 1, 'root', getCallSites(2));
       root.rootIntialize();
 
 const stateMap = new Map();
 let currState = root;
 
-function asyncInit(uid, handle, provider, parentUid) {
+function asyncInit(uid, type, parentUid, handle) {
+  // process._rawDebug('init:' + uid, parentUid);
+
+  // Ignore our nextTick for the root duration
+  // TODO(Fishrock123): detect this better.
+  if (uid === 2) return
+
+  // if (type === 'Timeout') {
+  //   process._rawDebug('Timeout', uid, handle._idleTimeout)
+  // }
+
   // get parent state
-  const state = (parentUid === null ? currState : stateMap.get(parentUid));
+  // root is always UID 1
+  const state = (parentUid === 1 ? currState : stateMap.get(parentUid));
 
   // add new state node
-  stateMap.set(uid, state.add(handle));
+  // process._rawDebug('>>> UID:', uid)
+  stateMap.set(uid, state.add(handle, uid, type));
 }
 
 function asyncBefore(uid) {
+  // Ignore our nextTick for the root duration
+  // TODO(Fishrock123): detect this better.
```

Comment:
`stateMap` can be used as a set of all seen init events. That should allow us to ignore all before, after, and destroy events where the init event was "disabled" away.